src/useMediaQuery.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect, useState } from 'react';
+
+export function useMediaQuery({ query }: { query: string }): boolean {
+  const mql = window.matchMedia(query);
+  const [result, setResult] = useState<boolean>(mql.matches);
```

Comment:
typescript сам поймёт тип
```suggestion
  const [result, setResult] = useState(mql.matches);
```